lightning/src/ln/channelmanager.rs

```diff
@@ -1918,6 +1935,26 @@ macro_rules! emit_channel_ready_event {
 
 macro_rules! handle_monitor_update_completion {
 	($self: ident, $peer_state_lock: expr, $peer_state: expr, $per_peer_state_lock: expr, $chan: expr) => { {
+		// If the channel belongs to a batch funding transaction, the progress of the batch
+		// should be updated as we have received funding_signed and persisted the monitor.
+		let mut completed_batch = None;
+		{
+			let v1_funding_batch_states = $self.v1_funding_batch_states.read().unwrap();
```

Comment:
Do you mean moving to `Mutex<HashMap<Txid, HashMap<([u8;32], PublicKey), bool>>>` for the state?

With regard to identifying channels as belonging to a batch, I chose to do the identification based on the groups of unbroadcasted transaction ID, as the `WaitingForBatch` flag is only set on `FundingSent` and I didn't want to have a redundant boolean on `Channel`.

Reply:
I originally checked whether a channel is part of a batch by looking at the transaction ID in this `HashMap` to reduce the amount of state (the `WaitingForBatch` flag is only set on the `FundingSent` state). This check occurred on receiving `funding_signed` and on monitor update completion for all channels.

This can also be done by keeping an indicator on `Channel` however which I've done now and moved to a single lock.